test/functional/p2p_fingerprint.py

```diff
@@ -121,14 +108,18 @@ def run_test(self):
         node0.sync_with_ping()
 
         # Request for very old stale block should now fail
+        with p2p_lock:
+            node0.last_message.pop("block", None)
```

Comment:
I think this can replace the section "# Send getdata & getheaders to refresh last received getheader message", no?

Reply:
Indeed, done. The "node0.sync_with_ping()" is still needed though to ensure that the newly generated block in the previous section "Longest chain is extended so stale is much older than chain tip" has been already received and doesn't interfere with the request for the old stale block test.